schema/gen/go/testStructsContainingMaybe_test.go

```diff
@@ -335,14 +40,119 @@ func TestStructsContainingMaybe(t *testing.T) {
 		}),
 	))
 
+	// There's a lot of cases to cover so a shorthand labels each case:
+	//  - 'v' -- value in that entry
+	//  - 'n' -- null in that entry
+	//  - 'z' -- absent entry
+	specs := []testcase{
+		{
+			name:     "all fields set -- vvvvv",
+			typeJson: `{"f1":"a","f2":"b","f3":"c","f4":"d","f5":"e"}`,
+			reprJson: `{"r1":"a","r2":"b","r3":"c","r4":"d","f5":"e"}`,
+			typePoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"f1", "a"},
+				{"f2", "b"},
+				{"f3", "c"},
+				{"f4", "d"},
+				{"f5", "e"},
+			},
+			reprPoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"r1", "a"},
+				{"r2", "b"},
+				{"r3", "c"},
+				{"r4", "d"},
+				{"f5", "e"},
+			},
+		},
+		{
+			name:     "setting nulls -- vvnnv",
+			typeJson: `{"f1":"a","f2":"b","f3":null,"f4":null,"f5":"e"}`,
+			reprJson: `{"r1":"a","r2":"b","r3":null,"r4":null,"f5":"e"}`,
+			typePoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"f1", "a"},
+				{"f2", "b"},
+				{"f3", ipld.Null},
+				{"f4", ipld.Null},
+				{"f5", "e"},
+			},
+			reprPoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"r1", "a"},
+				{"r2", "b"},
+				{"r3", ipld.Null},
+				{"r4", ipld.Null},
+				{"f5", "e"},
+			},
+		},
+		{
+			name:     "absent optionals -- vzvzv",
+			typeJson: `{"f1":"a","f3":"c","f5":"e"}`,
+			reprJson: `{"r1":"a","r3":"c","f5":"e"}`,
+			typePoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"f1", "a"},
+				{"f2", ipld.Absent},
+				{"f3", "c"},
+				{"f4", ipld.Absent},
+				{"f5", "e"},
+			},
+			reprPoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"r1", "a"},
+				//{"r2", ipld.ErrNotExists{}}, // TODO: need better error typing from traversal package.
+				{"r3", "c"},
+				//{"r4", ipld.ErrNotExists{}}, // TODO: need better error typing from traversal package.
+				{"f5", "e"},
+			},
+			typeItr: []entry{
+				{"f1", "a"},
+				{"f2", ipld.Absent},
+				{"f3", "c"},
+				{"f4", ipld.Absent},
+				{"f5", "e"},
+			},
+		},
+		{
+			name:     "absent trailing optionals -- vvnzz",
+			typeJson: `{"f1":"a","f2":"b","f3":null}`,
+			reprJson: `{"r1":"a","r2":"b","r3":null}`,
+			typePoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"f1", "a"},
+				{"f2", "b"},
+				{"f3", ipld.Null},
+				{"f4", ipld.Absent},
+				{"f5", ipld.Absent},
+			},
+			reprPoints: []testcasePoint{
+				{"", ipld.ReprKind_Map},
+				{"r1", "a"},
+				{"r2", "b"},
+				{"r3", ipld.Null},
+				//{"r4", ipld.ErrNotExists{}}, // TODO: need better error typing from traversal package.
+				//{"f5", ipld.ErrNotExists{}}, // TODO: need better error typing from traversal package.
+			},
+			typeItr: []entry{
+				{"f1", "a"},
+				{"f2", "b"},
+				{"f3", ipld.Null},
```

Comment:
This is weird, when compared to the previous case, but is it just supposed to be testing something about the guts of serialization (still a `TODO` I see), maybe this could do with a doc to explain why it's a special case—cause I can't really guess why it's this way.

Reply:
It's [covered in the docs on the testcase struct](https://github.com/ipld/go-ipld-prime/pull/66/files#diff-0e13278986646211b49e3940c88de7f4R45-R47): not all values are marshallable when examined at the type level (even though their `.Representation()` node of course is (and must be) marshallable).  Maps with complex keys, and structs with absent values, are both examples of this.  So for these we still want to test iterators, but can't use marshalling as a convenient way to do so: so we do something explicit instead.

Reply:
Followup: added the code to properly exercise this bit.  And yes, you did spy a feature that was wrong here.  Fixed :)